File: app/src/main/java/com/hhh/paws/database/dao/ProcedureDao.java
package com.hhh.paws.database.dao;

import com.hhh.paws.database.model.SurgicalProcedure;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

public interface ProcedureDao {
    Observable<List<SurgicalProcedure>> getAllProcedures(String petName);
    boolean setProcedure(String petName, SurgicalProcedure procedure);
    void deleteProcedure(String petName, SurgicalProcedure procedure);
}

File: app/src/main/java/com/hhh/paws/database/dao/ProcedureDao.kt
package com.hhh.paws.database.dao

import com.hhh.paws.database.model.SurgicalProcedure
import com.hhh.paws.util.UiState

interface ProcedureDao {
    suspend fun getAllProcedures(petName: String, result: (UiState<List<SurgicalProcedure>>) -> Unit)
    suspend fun setProcedure(petName: String, procedure: SurgicalProcedure, result: (UiState<String>) -> Unit)
    suspend fun deleteProcedure(petName: String, procedure: SurgicalProcedure, result: (UiState<String>) -> Unit)
}